validate_line

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+
+base_results_file=""
+
+usage()
+{
+	echo later
+	exit 0
+}
+
+error_out()
+{
+	echo $4
+	echo $3
+	echo field index checking: $field_index
+	echo field type: $field_type
+	exit 1
+}
+
+validate_line()
+{
+	re='^[0-9]+$'
+	line=$1
+	field_index=`echo $2 | cut -d':' -f 1`
+	field_type=`echo $2 | cut -d':' -f 2`
+
+	value=`echo $1 | cut -d':' -f $field_index | sed "s/ //g"`
+	if [[ $value == "" ]]; then
+		error_out $field_index $field_type $line "Either field is not present, or has no value."
+	fi
+	if [[ $field_type == "n" ]]; then
+		#
+		# Handle floats also.
+		#
+		check_value=`echo $value | sed "s/\./0/g"`
+		if ! [[ $check_value =~ $re ]] ; then
+			error_out $field_index $field_type $line "Non numerical field"
+		fi
+	fi
+	if [[ $field_type == "sm" ]]; then
+		#
+		# String, needs to match.
+		#
+		grep -q ^${value}: $base_results_file
+		if [ $? -ne 0 ]; then
+			error_out $field_index $field_type $line "Does not have match in the results file"
+		fi
+	fi
+	if [[ $field_type == "s" ]]; then
+		#
+		# Just a simple string.
+		#
+		grep -q ^${value}: $base_results_file
+		if [[ $value == "" ]]; then
+			error_out $field_index $field_type $line "Should be string value in the results file"
+		fi
```

Comment:
Maybe provide a regex value that you just send straight into grep.  Allows some flexibility in future.